### ISSUE TITLE
Make setproctitle optional

### ISSUE THREAD

**Issue by darikg** [dbcli/pgcli#392]
Installing on windows after #355 gives 

```
  Running setup.py install for setproctitle
    Complete output from command C:\Users\dg\Miniconda\envs\pgcli\python.exe -c "import setuptools, tokenize;__file__='c:\\users\\dg\\appdata\\local\\temp\\pip-build-aa9gjz\\setproctitle\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record c:\users\dg\appdata\local\temp\pip-rj1x5r-record\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_ext
    building 'setproctitle' extension
    error: Microsoft Visual C++ 9.0 is required (Unable to find vcvarsall.bat). Get it from http://aka.ms/vcpython27
```

Which is a little unfortunate

**Comment by amjith** [comment]
> Which is a little unfortunate

A little unfortunate? That's a huge inconvenience. I'm considering reverting the change. 

Is the feature even applicable in Windows? 

@darikg When you run the old version of pgcli (0.19.2) how does it show up in the Task Manager? Does the password show up in clear text?

If it doesn't we can make that package optional in Windows and not do the proc title rewriting.

**Comment by darikg** [comment]
Good point. It looks like it just shows up as `pgcli.exe` so there's no need to play with the process title at all.

**Comment by amjith** [comment]
Fix released in 0.20.0. 

Please upgrade:

```
pip install -U pgcli
```

**Comment by asfaltboy** [comment]
Can we really make `setproctitle` optional, not just on windows? For me the trouble was needing gcc in a sterile linux environment. I would open a new issue, but the request is the same, make `setproctitle` optional.

As possible solutions, I would either add it in [an extra][1] or, if including it by default is necessary, look into adding an option to disable it during installation (e.g using an environment variable such as `NO_SETPROCTITLE=1` which would also exclude it from requirements.

[1]: http://setuptools.readthedocs.io/en/latest/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies
